hyperspy/model.py

```diff
@@ -609,12 +456,12 @@ def as_signal(self, component_list=None, out_of_range_to_nan=True,
         pbar.finish()
```

Comment:
Here slightly higher (can't comment there for some reason) it should read 
```python
data[self.axes_manager._getitem_tuple][
    self.channel_switches] = self.__call__(
    non_convolved=not self.convolved, onlyactive=True).ravel()
```
(i.e. add `.ravel()` to the end)